### PR TITLE
Added Abs to as_real_imag

### LINKED ISSUE #1381: Add Abs to as_real_imag

### PR DESCRIPTION
**Pull request by npochhi**
Fixes #1381 
Should I add more tests?

### CODE REVIEW

File: symengine/tests/basic/test_as_real_imag.cpp [hunk 1]
```diff
@@ -259,3 +260,17 @@ TEST_CASE("RealImag: Trigonometric functions", "[as_real_imag]")
...
+TEST_CASE("RealImag: Absolute Value Function", "[as_real_imag]")
+{
+    RCP<const Basic> re, im;
+    auto i2 = integer(2), i3 = integer(3);
+
+    as_real_imag(abs(add(add(i2, I), add(i3, I))), outArg(re), outArg(im));
+    REQUIRE(eq(*re, *sqrt(add(pow(add(i2, i3), i2), pow(i2, i2)))));
+    REQUIRE(eq(*im, *zero));
+
+    as_real_imag(abs(neg(i2)), outArg(re), outArg(im));
```
Comment (isuruf) [inline symengine/tests/basic/test_as_real_imag.cpp]: These are numeric functions and `abs(-2)` is automatically simplified to `2`. Use a symbol inside `abs` like `abs(x-2)`

File: symengine/as_real_imag.cpp [hunk 2]
```diff
@@ -221,6 +221,18 @@ class RealImagVisitor : public BaseVisitor<RealImagVisitor>
...
 
+    void bvisit(const Abs &x)
+    {
+        apply(*x.get_arg());
+        if (eq(**imag_, *zero)) {
+            *real_ = x.rcp_from_this();
+            return;
+        }
+        auto i2 = integer(2);
+        *real_ = sqrt(add(pow(*real_, i2), pow(*imag_, i2)));
```
Comment (isuruf) [inline symengine/as_real_imag.cpp]: This should just be `real_ = x`

File: symengine/tests/basic/test_as_real_imag.cpp [hunk 3]
```diff
@@ -259,3 +260,17 @@ TEST_CASE("RealImag: Trigonometric functions", "[as_real_imag]")
...
+    REQUIRE(eq(*re, *abs(neg(i2))));
+    REQUIRE(eq(*im, *zero));
+}
```
Comment (ranjithkumar007) [inline symengine/tests/basic/test_as_real_imag.cpp]: minor one: Add new line.

**Comment by npochhi** [comment]
ping @isuruf

**Pull request merged**